gammapy/maps/geom.py

```diff
@@ -904,6 +902,33 @@ def __repr__(self):
         str_ += "\tcoordsys : {}\n".format(self.coordsys)
         return str_
 
+    #TODO: this is a temporary solution until we have decided how to handle
+    # quantities uniformly. This should be called after any `MapCoord.create()`
+    # to support that users can pass quantities in any Map.xxx_by_coord() method.
+    def match_axes_units(self, geom):
+        """Match the units of the non-spatial axes to a given map geometry.
+
+        Parameters
+        ----------
+        geom : `MapGeom`
+            Map geometry with specified units per axis.
+
+        Returns
+        -------
+        coords : `MapCoord`
+            Map coord object with matched units
+        """
+        coords = {}
```

Comment:
Is this guaranteed to work on Python 2, or could order be a problem?

Maybe change to OrderedDict here, or make a list of (key, val) and pass that to OrderedDict on the last line?